src/services/user.service.js

```diff
@@ -9,98 +9,100 @@ exports.getByUUID = async (uuid) => User.findOne({ uuid });
-    const validPass = await bcrypt.compare(password, user.password);
-    if (!validPass) reject(new Error('Invalid password'));
+  return { accessToken, refreshToken, expires_in: process.env.ACCESS_TOKEN_TTL };
+};
 
-    const accessToken = jwt.sign(
-      { uuid: user.uuid },
-      process.env.ACCESS_TOKEN_SECRET,
-      { expiresIn: process.env.ACCESS_TOKEN_TTL },
-    );
-    const refreshToken = jwt.sign(
-      { uuid: user.uuid },
-      process.env.REFRESH_TOKEN_SECRET,
-      { expiresIn: process.env.REFRESH_TOKEN_TTL },
-    );
+exports.login = async (username, password) => {
+  const user = await getByUsername(username);
 
-    resolve({ accessToken, refreshToken, expires_in: process.env.ACCESS_TOKEN_TTL });
-  }).catch(() => {
-    reject(new Error('User does not exist'));
-  });
-});
+  if (user == null) {
+    throw new Error('User does not exist');
+  }
 
-exports.me = async (uuid) => new Promise((resolve, reject) => {
-  getByUUID(uuid).then((user) => {
-    resolve({
+  const validPass = await bcrypt.compare(password, user.password);
+  if (!validPass) throw new Error('Invalid password');
+
+  const accessToken = jwt.sign(
+    { uuid: user.uuid },
+    process.env.ACCESS_TOKEN_SECRET,
+    { expiresIn: process.env.ACCESS_TOKEN_TTL },
+  );
+  const refreshToken = jwt.sign(
+    { uuid: user.uuid },
+    process.env.REFRESH_TOKEN_SECRET,
+    { expiresIn: process.env.REFRESH_TOKEN_TTL },
+  );
+
+  return { accessToken, refreshToken, expires_in: process.env.ACCESS_TOKEN_TTL };
+};
+
+exports.me = async (uuid) => {
+  try {
+    const user = await getByUUID(uuid);
+    return {
       username: user.username,
       uuid: user.uuid,
       registerTimestamp: user.registerTimestamp,
       playedGames: user.playedGames,
-    });
-  }).catch((err) => {
-    reject(new Error(err));
-  });
-});
+    };
+  } catch (err) {
+    throw new Error(err);
+  }
+};
 
-exports.refresh = async (refreshToken) => new Promise((resolve, reject) => {
+exports.refresh = async (refreshToken) => {
   let verifiedRefreshToken;
   try {
     verifiedRefreshToken = jwt.verify(
       refreshToken,
       process.env.REFRESH_TOKEN_SECRET,
     );
   } catch (error) {
-    reject(new Error('Invalid refreshToken'));
+    throw new Error('Invalid refreshToken');
   }
   const accessToken = jwt.sign(
     { uuid: verifiedRefreshToken.uuid },
     process.env.ACCESS_TOKEN_SECRET,
     { expiresIn: process.env.ACCESS_TOKEN_TTL },
   );
-  resolve({ accessToken });
-});
+  return { accessToken };
+};
 
-exports.setWebsocket = async (uuid, socketID) => new Promise((resolve, reject) => {
-  getByUUID(uuid).then((user) => {
+exports.setWebsocket = async (uuid, socketID) => {
+  try {
+    const user = await getByUUID(uuid);
     user.websocket = socketID;
-    user.save();
-    resolve('websocket set');
-  }).catch((err) => {
+    await user.save();
+    return 'websocket set';
+  } catch (err) {
     // TODO: Dont throw error since it kills the server
```

Comment:
To prevent the backend from crashing, we could just remove the `throw` or catch the error in the `socket.service`, but maybe we want to add a retry or some error propagation to notify the user. Do you have any opinions/ideas here?

Reply:
It is unlikely that we can recover from this error or send a message to the user. We could add a retry, but I will cust convert the throw to a log for now